src/test/java/io/reactivex/schedulers/SchedulerTest.java

```diff
@@ -307,4 +307,66 @@ public void customScheduleDirectDisposed() {
 
         assertTrue(d.isDisposed());
     }
+
+    @Test
+    public void unwrapDefaultPeriodicTask() {
```

Comment:
Shouldn't we move these tests to `AbstractSchedulerTests` @akarnokd?

Reply:
So I'm waiting for your decision :)

Reply:
While we're waiting for @akarnokd to answer I'd recommend you to try to move this code to `AbstractSchedulerTests` and make sure to use `getScheduler()` instead of `TestScheduler`, that way we check that all schedulers properly support this introspection

Reply:
It's fine here.

Reply:
Don't we want to check that all schedulers support this API?

Reply:
It can get complicated and this is an optional introspection. If @lukaszguz wants to expand it, then sure.

Reply:
True, but I'd encourage @lukaszguz to try that and ask for help if needed, would be great to provide as robust API experience as possible

Reply:
Maybe it's enough for now? I can always add a new PR with additional tests.

Reply:
I'm fine with this as is. We can always expand the feature set later.

Reply:
This is my first time when I'm doing PR to your library. Thanks to your tips, I know more or less what your requirements are and I can write better code. I think the better option would be to leave those tests in `SchedulerTest` because they are showed how `Scheduler` works but also good idea is to add tests in `AbstractSchedulerTests` which they will guarantee compatibility with all rx's schedulers